### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by constantinpape**

### CODE REVIEW

File: utils/data_loader.py [hunk 1]
```diff
@@ -1,24 +1,46 @@
 from torch.utils.data import DataLoader
-from .heatmap_loader import HeatmapLoader
+from .heatmap_dataset import HeatmapDataset
 
-def CreateDataLoader(train_images, train_labels, val_images, val_labels, test_images, test_labels, raw_transform, transform, patch_shape, num_workers, batch_size, eps=0.00001, sigma=None, lower_bound=None, upper_bound=None):
```
Comment (constantinpape) [inline utils/data_loader.py:4]: Function names should always be snake_case, i.e. `create_data_loader`, instead of CamelCalse, `CreateDataLoader`, in python.

File: utils/heatmap_dataset.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 import numpy as np
 
 import zarr
 import os
 
 from data_processing.create_heatmap import process_tomogram
 
-class HeatmapLoader(torch.utils.data.Dataset):
+
+class HeatmapDataset(torch.utils.data.Dataset):
```
Comment (constantinpape) [inline utils/heatmap_dataset.py:12]: This is a Dataset, not a Dataloader, so it should be named accordingly. See e.g. https://pytorch.org/tutorials/beginner/basics/data_tutorial.html for the distinction.

Comment (constantinpape) [reply utils/heatmap_dataset.py:12]: Also, we should refactor this loader a bit, but we can discuss how to do this later. I think the best option would be to inherit from https://github.com/constantinpape/torch-em/blob/main/torch_em/data/segmentation_dataset.py .

File: utils/heatmap_dataset.py [hunk 3]
```diff
@@ -49,6 +50,8 @@ def __init__(
         self.lower_bound = lower_bound
         self.upper_bound = upper_bound
 
+        # TODO: n_samples should be derived from how often the bounding box fits
+        # rather than the number of volumes (as in SegmentationDataset)
```
Comment (constantinpape) [inline utils/heatmap_dataset.py:54]: The logic for doing this is here: https://github.com/constantinpape/torch-em/blob/main/torch_em/data/segmentation_dataset.py#L53

File: utils/heatmap_dataset.py [hunk 4]
```diff
@@ -73,18 +76,31 @@ def _sample_bounding_box(self, shape):
...
             index = np.random.randint(0, len(self.raw_images))
         raw, label = self.raw_images[index], self.label_images[index]
 
-        #TODO this is specific for challenge zarr files now, maybe need to generalize in the future
-        #zarr_file = zarr.open(f"{raw}/VoxelSpacing10.000/denoised.zarr/0", mode='r')
+        # TODO this is specific for challenge zarr files now, maybe need to generalize in the future
+        # Yes ;). We can discuss this soon.
+
+        # zarr_file = zarr.open(f"{raw}/VoxelSpacing10.000/denoised.zarr/0", mode='r')
         zarr_file = zarr.open(os.path.join(raw, "VoxelSpacing10.000", "denoised.zarr", "0"), mode='r')
```
Comment (constantinpape) [inline utils/heatmap_dataset.py:89]: The best option here is to use the full path to the zarr file as image path (i.e. the file ending in `"... .zarr"`), and then use `raw_key="0"`.

File: utils/heatmap_dataset.py [hunk 5]
```diff
@@ -73,18 +76,31 @@ def _sample_bounding_box(self, shape):
...
-        raw = zarr_file[:]
-        #sigma is not really used in my process_tomogram ... TODO ?
-        label = process_tomogram(label, raw.shape, eps=self.eps, sigma=self.sigma, lower_bound=self.lower_bound, upper_bound=self.upper_bound)
+
+        # This was very inefficient!
+        # You first load the full data from zarr and then later load the bounding box.
+        # raw = zarr_file[:]
+        # Instead, you can just load the bounding box from the zarr
+        raw = zarr_file
```
Comment (constantinpape) [inline utils/heatmap_dataset.py:95]: Your previous code was quite inefficient because it always loaded the full data from the zarr file, although we just need to load the bounding box.

File: utils/heatmap_dataset.py [hunk 6]
```diff
@@ -73,18 +76,31 @@ def _sample_bounding_box(self, shape):
...
+
+        # This is also quite inefficient.
+        # You compute he labels for the full tomogram, and then sub-sample.
+        # sigma is not really used in my process_tomogram ... TODO ?
+        label = process_tomogram(
+            label, raw.shape, eps=self.eps, sigma=self.sigma,
+            lower_bound=self.lower_bound, upper_bound=self.upper_bound
+        )
```
Comment (constantinpape) [inline utils/heatmap_dataset.py:103]: This is quite inefficient, because you create the full label volume, which involves a gaussian for each protein type over the volume. But we only need this for the bounding box.

File: utils/training.py [hunk 7]
```diff
@@ -69,14 +71,17 @@ def supervised_training(
...
+    augmentations: Optional[bool] = False,
+    eps: float = 1e-5,
+    sigma: int = None,
+    lower_bound: float = None,
+    upper_bound: float = None,
     test_paths: Optional[Tuple[str]] = None,
     test_label_paths: Optional[Tuple[str]] = None,
     save_root: Optional[str] = None,
+    n_samples_train: Optional[int] = None,
+    n_samples_val: Optional[int] = None,
```
Comment (constantinpape) [inline utils/training.py:83]: I have added the samples to control the length of an epoch.

File: utils/training.py [hunk 8]
```diff
@@ -69,14 +71,17 @@ def supervised_training(
...
+    dataset_class=HeatmapDataset,
```
Comment (constantinpape) [inline utils/training.py:84]: I have made the dataset class more flexible, so that I can use this for my own training.

**Review by constantinpape — COMMENTED** [review]
Hi @SarahMuth ,
this contains many changes to make my linter happy. I commented on any change that is a bit more relevant, or where we should eventually update the code (we can discuss these parts in person). Just for reference, here is my code that uses this for a 3D point detection training: https://github.com/computational-cell-analytics/flamingo-tools/tree/syn-marker-training/scripts/synapse_marker_detection
